### PR TITLE
Move CTE related tests out of TINC

### PR DESCRIPTION
**Pull request by dhanashreek89**
Move CTE tests out of TINC and add (the ones that are not already present) to main test suite. 
optimizer_functional_part1 runs 58 CTE tests which can be categorized as :

- cte_queries_<1-24>

Almost all these tests (with one exception as described next) are already in the main test suite in `qp_with_clause.sql` which means we had duplicate tests. Hence this commit removes them from TINC.

Two queries in `cte_queries24.sql`, query#6 and query#11 were missing in `qp_with-clause.sql`. This commit adds them.

- cte_functest_<58-60>

In the old commit 629f7e7, these tests were not moved since they produced different results on different invocations andwere marked as skipped tests. They indeed have different behavior with
Planner and ORCA. ORCA does not support updating rows when there are multiple matches in the join condition and errors out. Planner on the other hand allows this but the results are non-deterministic. There is already coverage for this error in `qp_dml_joins.sql` and `DML_over_joins.sql`. Hence this PR drops these tests. 

- enable_cte_plan_space

This is a pure ORCA test which tests the number of plan alternatives produced by ORCA when optimizer_cte_inlining is ON and OFF. This will be moved in ORCA test suite (see https://github.com/greenplum-db/gporca/pull/252)

- icg_cte_with_values

The test is already present in `with.sql`.

- cte_functest_<22-23>_inlining_<enabled/disabled>

Moved these tests to `qp_with_functional.sql`. qp_with_functional is tested with both inlining and noinlining.

This commit thus removes the `cte` test folder from TINC

### CODE REVIEW

File: src/test/regress/sql/qp_with_functional.sql [hunk 1]
```diff
@@ -746,6 +780,118 @@ SELECT * FROM newfoo;
 
 DROP TABLE newfoo;
 
+-- Tests 27d, 27e and 27f below have different behaviours with legacy optimizer and ORCA
+-- ORCA does not support updating rows when there are multiple matches in the join condition.
+-- Planner however allows this and the results are non-deterministic.
+-- Hence these tets are run with `optimizer=on`.
+ 
+SET optimizer=ON;
```
Comment (vraghavan78) [inline src/test/regress/sql/qp_with_functional.sql]: -1 on this pattern. This means that these tests won't be exercised with planner, and these will error out when the OSS developer does not build with Orca.

Here are my suggestions:
* If Planner and Orca behavior is different and is expected to be, then have a different answer file
* If Planner errors out or crashes then disable it and open an issue on gpdb. We should fix it. Comment out the test and may be add it to gporca as a unit test

File: src/test/regress/expected/qp_with_functional_noinlining.out [hunk 2]
```diff
@@ -1328,6 +1366,267 @@ SELECT * FROM newfoo;
 (0 rows)
 
 DROP TABLE newfoo;
+-- Tests 27d, 27e and 27f below have different behaviours with legacy optimizer and ORCA
+-- ORCA does not support updating rows when there are multiple matches in the join condition.
+-- Planner however allows this and the results are non-deterministic.
+-- Hence these tets are run with `optimizer=on`.
```
Comment (vraghavan78) [inline src/test/regress/expected/qp_with_functional_noinlining.out]: tets -> test

**Comment by dhanashreek89** [comment]
@vraghavan78 Fair point. Thanks !
IMO we should remove these tests. Here is why:
- These tests show different behavior with Planner and ORCA.  ORCA does not support updating rows when there are multiple matches in the join condition. However Planner allows this and the results produced are  non-deterministic. Here is the output of two consecutive runs with Planner:
```sql
Run 1:

SELECT * FROM newfoo ORDER BY 1;
 a  | b
----+----
  1 |  2
  2 |  3
  3 |  4
  4 |  5
  5 |  6
  6 |  7
  7 |  8
  8 |  9
  9 | 10
 10 | 11
(10 rows)

WITH CTE(c,d) as
        (
                SELECT a,b FROM foo WHERE a > 1
        )
SELECT CTE.c,bar.d FROM  CTE,bar where CTE.c = bar.d ORDER BY 1;
 c  | d
----+----
  2 |  2
  3 |  3
  4 |  4
  5 |  5
  6 |  6
  7 |  7
  8 |  8
  9 |  9
 10 | 10
(9 rows)

UPDATE newfoo SET b =sub.c FROM
(
        WITH CTE(c,d) as
        (
                SELECT a,b FROM foo WHERE a > 1
        )
        SELECT CTE.c,bar.d FROM  CTE,bar where CTE.c = bar.d ORDER BY 1
) sub;
SELECT * FROM newfoo ORDER BY 1;
 a  | b
----+---
  1 | 2
  2 | 2
  3 | 8
  4 | 8
  5 | 8
  6 | 8
  7 | 8
  8 | 8
  9 | 8
 10 | 8
(10 rows)


Run 2:
SELECT * FROM newfoo ORDER BY 1;
 a  | b
----+----
  1 |  2
  2 |  3
  3 |  4
  4 |  5
  5 |  6
  6 |  7
  7 |  8
  8 |  9
  9 | 10
 10 | 11
(10 rows)

WITH CTE(c,d) as
        (
                SELECT a,b FROM foo WHERE a > 1
        )
SELECT CTE.c,bar.d FROM  CTE,bar where CTE.c = bar.d ORDER BY 1;
 c  | d
----+----
  2 |  2
  3 |  3
  4 |  4
  5 |  5
  6 |  6
  7 |  7
  8 |  8
  9 |  9
 10 | 10
(9 rows)

UPDATE newfoo SET b =sub.c FROM
(
        WITH CTE(c,d) as
        (
                SELECT a,b FROM foo WHERE a > 1
        )
        SELECT CTE.c,bar.d FROM  CTE,bar where CTE.c = bar.d ORDER BY 1
) sub;
SELECT * FROM newfoo ORDER BY 1;
 a  | b
----+---
  1 | 2
  2 | 2
  3 | 2
  4 | 2
  5 | 2
  6 | 2
  7 | 2
  8 | 2
  9 | 2
 10 | 2
(10 rows)
```
This is a known behavior for both Planner and ORCA  hence there is nothing actionable here. So adding new answer file for planner is not an option.

- Adding a test in ORCA will not be useful since it will not test the error (being thrown by executor).

- we already have coverage for this behavior (error generated by ORCA) at other places such as `DML_over_joins.sql` and `qp_dml_joins.sql`

Thoughts?

**Comment by vraghavan78** [comment]
Sounds good.

**Pull request merged**